src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DockerImageDownloader.java

```diff
@@ -276,4 +285,47 @@ private <T> T runWithConnectionErrorCheck(CrashableSupplier<T, Exception> task)
             throw e;
         }
     }
+
+    /**
+     * Cleanup component, delete docker image when component being removed.
+     * @throws PackageLoadingException from getPackageRecipe
+     */
+    @Override
+    public void cleanup() throws PackageLoadingException {
+        // this docker image not only used by itself
+        if (!ifImageUsedByOther(componentStore)) {
+            try {
+                Image image = DockerImageArtifactParser
+                        .getImage(ComponentArtifact.builder().artifactUri(artifact.getArtifactUri()).build());
+                dockerClient.deleteImage(image);
+            } catch (InvalidArtifactUriException | DockerImageDeleteException e) {
+                logger.atWarn().kv("docker image", artifact.getArtifactUri())
```

Comment:
this error should be thrown up to the caller and logged and handled by the caller.